utils/configuration_store_generator/configuration_store_structure_generator.py

```diff
@@ -0,0 +1,251 @@
+        # TODO create bounds set which will reduce the number of bounds
+        return f"MemConfigItem<{self.get_type_name()}> {self.get_name()} {{ \"{self.name}\" , {str(self.default)} }};"
+
+    @abstractmethod
+    def get_additional_definitions(self) -> Optional[str]:
+        pass
+
+    @abstractmethod
+    def get_type_name(self) -> str:
+        pass
+
+    def get_name(self) -> str:
+        return self.name
+
+    @property
+    @abstractmethod
+    def default(self) -> str:
+        pass
+
+
+class BasicItem(Item):
+    data_type: DataType
+    def_val: Any
+
+    def __init__(self, name: str, data_type: DataType, default: Any):
+        self.name = name
+        self.data_type = data_type
+        self.def_val = default
+
+    def get_additional_definitions(self) -> Optional[str]:
+        return None
+
+    def get_type_name(self):
+        return self.data_type.value
+
+    @property
+    def default(self) -> str:
+        if type(self.def_val) == bool:
+            if self.def_val:
+                return "true"
+            else:
+                return "false"
+        else:
+            return str(self.def_val)
+
+
+@dataclass
+class ArrayItem(Item):
+    size: int
+    item: Item
+
+    def get_additional_definitions(self) -> Optional[str]:
+        return None
+
+    def get_type_name(self):
+        return f"std::array<{self.item.get_type_name()},{self.size}>"
+
+    @property
+    def default(self) -> str:
+        return self.item.default
+
+
+@dataclass
+class StructItem(Item):
+    items: dict
+    type_name: str
+
+    def __get_members(self):
+        members = ""
+        for name, item in self.items.items():
+            members = members + f"{item.get_type_name()} {name};\n"
+        return members
+
+    def get_type_name(self):
+        return self.type_name
+
+    def get_additional_definitions(self) -> Optional[str]:
+        return f"""struct {self.get_type_name()} {{
+                    {self.__get_members()}
+        }};"""
+
+    @property
+    def default(self) -> str:
+        bound = "{ "
+        for item_name, item in self.items.items():
+            bound = bound + f"{item.default}, "
+        return str(bound[0:-2]) + " }"
+
+
+class ItemParser:
+    bounds: dict
+    items: dict
+    additional: list
+
+    def __init__(self):
+        self.bounds = dict()
+        self.items = dict()
+        self.additional = []
+
+    def __create_basic_item(self, name: str, data_type: str,
+                            default: Any) -> Item:
+        data_type = DataType[data_type]
+        return BasicItem(name, data_type, default)
+
+    def __create_array_item(self,
+                            name: str,
+                            data_type: str,
+                            length: int,
+                            item: dict,
+                            default=None) -> Item:
+        member = self.__parse_type("item", item)
+
+        return ArrayItem(name, length, member)
```

Comment:
It seems to me that you can't create an array of N elements and set the default to N distinctive values. Is that intentional? 🤔

Reply:
Yes, I havent found reason to do have different defaults. But if tyou have some I will add the posibility to add different defaults. 

Strings have their own type.